bark/world/evaluation/rss/safety_polygon.hpp

```diff
@@ -102,24 +118,39 @@ inline void ComputeSafetyPolygon(
       bg::set<1>(pt, y_new);
     }
 
-    // calculate if the agent is in front or behind
-    auto other_agent = observed_world.GetAgents()[safe_poly.agent_id];
-    auto other_pose = other_agent->GetCurrentPosition();
-    double relative_angle = atan2(
-      bg::get<1>(ego_pose) - bg::get<1>(other_pose),
-      bg::get<0>(ego_pose) - bg::get<0>(other_pose));
-    double diff_angle = SignedAngleDiff(theta - M_PI_2, relative_angle);
-    double sgn_lon_in_front = diff_angle > 0 ? 1 : -1;
-
-    // this enables directional computation for the longitudinal
-    // safety distance
-    // if the signs are different, sgn_lon is set to zero
-    // Thus, the original point of the polygon will not be modified
-    // NOTE: often the safety distance is returned as 1+e9
-    if (sgn_lon_in_front*sgn_lon < 0. ||
-        safe_poly.lon_safety_distance > 10000.)
-      sgn_lon = 0.;
+    // draws the polygon in relation to the other agent
+    if (directional == LonDirectionMode::AUTO) {
+      // calculate if the agent is in front or behind
+      auto other_agent = observed_world.GetAgents()[safe_poly.agent_id];
+      if (!other_agent)
+        return;
+      auto other_pose = other_agent->GetCurrentPosition();
+      double relative_angle = atan2(
+        bg::get<1>(ego_pose) - bg::get<1>(other_pose),
+        bg::get<0>(ego_pose) - bg::get<0>(other_pose));
+      double diff_angle = SignedAngleDiff(theta - M_PI_2, relative_angle);
```

Comment:
why M_PI_2? is it then always within [-pi, pi]?

Reply:
this is a 90 degree turn